numpy/_core/multiarray.py

```diff
@@ -434,11 +434,41 @@ def where(condition, x=None, y=None):
 
 
 @array_function_from_c_func_and_dispatcher(_multiarray_umath.lexsort)
```

Comment:
you should be using the `__array_function__` dispatcher for python functions, and rename this function to be `lexsort`